### PR TITLE
Add docker publish

### PR DESCRIPTION
**Pull request by mhaber-tophat**
builds and publishes a docker container on release for the sanity-client.

also updating example makefile with example of how we are setting VAR's in the config

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -21,25 +21,48 @@ commands:
...
             - node_modules/
-
 jobs:
   build:
     docker:
       - image: circleci/node:8.10.0
     working_directory: ~/repo
     steps:
       - checkout
+      - setup_remote_docker
```
Comment (jakebolam) [inline .circleci/config.yml]: Wheres this come from?

File: .circleci/config.yml [hunk 2]
```diff
@@ -21,25 +21,48 @@ commands:
...
       - run: make -C service install
       - run: make -C client package
       - run: make -C client release-version
       - run: make -C service package
+      - run: make -C client build-docker
+      - run:
+          name: "Cache Docker Image"
+          command: |
+            mkdir -p docker-cache
+            docker save -o docker-cache/sanity-docker.tar tophatmonocle/sanity-runner:latest
```
Comment (jakebolam) [inline .circleci/config.yml]: `tophatmonocle/sanity-runner:latest` -> `tophat/sanity-runner:latest`

File: .circleci/config.yml [hunk 3]
```diff
@@ -21,25 +21,48 @@ commands:
...
+    steps:
+      - setup_remote_docker
+      - attach_workspace:
+          at: ~/workspace
+      - run:
+          name: "Publish Release on Docker"
+          command: |
+            VERSION=`cat ~/workspace/client/version.txt`
+            docker load < ~/workspace/docker-cache/sanity-docker.tar
+            docker tag tophatmonocle/sanity-runner:latest tophatmonocle/sanity-runner:${VERSION}
```
Comment (jakebolam) [inline .circleci/config.yml]: tophatmonocle/sanity-runner:latest -> tophat/sanity-runner:latest

File: .circleci/config.yml [hunk 4]
```diff
@@ -21,25 +21,48 @@ commands:
...
+            echo ${DOCKER_PASSWORD} | docker login --username ${DOCKER_USERNAME} --password-stdin
+            docker push tophatmonocle/sanity-runner:${VERSION}
```
Comment (jakebolam) [inline .circleci/config.yml]: tophatmonocle/sanity-runner:latest -> tophat/sanity-runner:latest

File: client/Makefile [hunk 5]
```diff
@@ -21,5 +21,8 @@ check-versions:
 .PHONY: clean
 clean:
 	@rm -rf node_modules
-	
 
+.PHONY: build-docker
+build-docker:
+	@echo "Building Docker Container"
+	docker build . -t tophatmonocle/sanity-runner
```
Comment (jakebolam) [inline client/Makefile]: tophatmonocle/sanity-runner:latest -> tophat/sanity-runner:latest

File: .circleci/config.yml (hunk 1 above)
Comment (mhaber-tophat) [reply .circleci/config.yml]: setup_remote_docker is pre-defined circleci step. it gives you a remote environment that it then runs all the docker commands against. It makes it so you dont need set up a docker imag with a working docker daemon inside of it. its really nice. https://circleci.com/docs/2.0/building-docker-images/

**Pull request closed without merging**